src/magellan_vision/scripts/line_detector.py

```diff
@@ -1,211 +1,159 @@
+            print 'Subscribe to publish frequency: %s Hz' % (1/(time3-time0))
+
+
+# Class Lines: Detects points in an Image that aligns into a line. Outputs the image with detected points overlayed
+#              on it.
+class Lines(object):
+
+    def __init__(self):
+        self._ddepth = cv2.CV_16S
+        self._kernel_size = 3
+        self.points_arr = []
+
+    def skeletize(self, img, size, skel):
+        element = cv2.getStructuringElement(cv2.MORPH_CROSS, (3, 3))
+        done = False
+        while(not done):
+            eroded = cv2.erode(img, element)
+            temp = cv2.dilate(eroded, element)
+            temp = cv2.subtract(img, temp)
+            skel = cv2.bitwise_or(skel, temp)
+            img = eroded.copy()
+            zeros = size - cv2.countNonZero(img)
+            if zeros == size:
+                done = True
+        return skel
+
+    def detect(self, cv_image):
+
+        gray = cv2.cvtColor(cv_image, cv2.COLOR_BGR2GRAY)
+        blurred = cv2.GaussianBlur(gray, (3, 3), 0)
+        ret, thresh4 = cv2.threshold(blurred, 200, 255, cv2.THRESH_TOZERO)
+        size = np.size(thresh4)
+        skel = np.zeros(thresh4.shape, np.uint8)
+
+        skeleton = self.skeletize(thresh4, size, skel)
+
+        dst = cv2.Laplacian(skeleton, self._ddepth, ksize=self._kernel_size)
+        abs_dst = cv2.convertScaleAbs(dst)
+
+        linesP = cv2.HoughLinesP(abs_dst, rho=6, theta=np.pi / 60, threshold=50,
+                                 lines=np.array([]), minLineLength=30, maxLineGap=10)
+
+        if linesP is not None:
+            for line in linesP:
+                for x1, y1, x2, y2 in line:
+                    slope = (y2 - y1) / (x2 - x1)
+                    self.points_arr.append((x1, y1))
+                    self.points_arr.append((x2, y2))
+                    # <-- Calculating the slope.
+                    if math.fabs(slope) < .5:
+                        # <-- Only consider extreme slope
+                        continue
+                    if x1 < cv_image.shape[1]/2 and x2 < cv_image.shape[1]/2:
+                        # <-- If the slope is negative, left group
+                        cv2.circle(cv_image, (x1, y1), (5), (0, 0, 255), 3)
+                        cv2.circle(cv_image, (x2, y2), (5), (0, 0, 255), 3)
+                    else:  # <-- Otherwise, right group.
+                        cv2.circle(cv_image, (x1, y1), (5), (0, 255, 0), 3)
+                        cv2.circle(cv_image, (x2, y2), (5), (0, 255, 0), 3)
+        return cv_image
+
+
+def main(args):
+    # ROS Node Initialization
+    # disable_signals flag allows catching signals(Excecptions) such as the KeyboardInterrupt, otherwise try/except
+    # Exceptions may never be handled
+
+    rospy.init_node('LinesNode', anonymous=True, disable_signals=True)
+    PubSubNode()
+    try:
+        rospy.spin()
+    except KeyboardInterrupt:
+        print "shutting down ROS Lines detector Module"
+    cv2.destroyAllWindows()
```

Comment:
We shouldn't be creating any windows in this node so we shouldn't be calling this.

Reply:
fixed